main/src/ca/uwaterloo/flix/language/ast/WeededAst.scala

```diff
@@ -35,9 +35,9 @@ object WeededAst {
 
     case class Instance(doc: Ast.Doc, mod: Ast.Modifiers, clazz: Name.QName, tpe: WeededAst.Type, tconstrs: List[WeededAst.TypeConstraint], defs: List[WeededAst.Declaration.Def], loc: SourceLocation) extends WeededAst.Declaration
 
-    case class Sig(doc: Ast.Doc, ann: List[WeededAst.Annotation], mod: Ast.Modifiers, ident: Name.Ident, tparams: WeededAst.KindedTypeParams, fparams: List[WeededAst.FormalParam], exp: Option[WeededAst.Expression], tpe: WeededAst.Type, retTpe: WeededAst.Type, pur: WeededAst.Type, eff: WeededAst.EffectSet, tconstrs: List[WeededAst.TypeConstraint], loc: SourceLocation)
+    case class Sig(doc: Ast.Doc, ann: List[WeededAst.Annotation], mod: Ast.Modifiers, ident: Name.Ident, tparams: WeededAst.KindedTypeParams, fparams: List[WeededAst.FormalParam], exp: Option[WeededAst.Expression], tpe: WeededAst.Type, retTpe: WeededAst.Type, eff: WeededAst.Type, put: WeededAst.Type, tconstrs: List[WeededAst.TypeConstraint], loc: SourceLocation)
```

Comment:
put-> pur

Reply:
Also, I thought the order was that purity information is always before effect information?

Reply:
I'll go back and fix that. I was doing alphabetical :)

Reply:
Ok, but I am not sure now. How does the existing code look? I just thought it was always `pur`, `eff` because that seemed natural.

Reply:
The other place the pairing is used is in the `EffectOrPurity` structures in the parser. In Flix code they're effectively unordered, since you can't express both.